### PR TITLE
feat: add cmap subpackage

### PR DESCRIPTION
**Pull request by tlambert03**
this adds a new subpackage called `cmap` (have the name on pypi ... but also kinda like `klutz` 😄 can decide later)

For now, let's use it internally, but it feels very nice in interactive usage!

https://user-images.githubusercontent.com/1609449/209978070-127b137f-00f5-4a23-861c-6976dcd14dee.mov

see also `examples/colormaps.py` that shows how to use colormaps with matplotlib, napari, vispy, pygfx, bokeh, and altair

:eyes: one thing that is lost here is the autogenerated widget in the controller... but a nice magicgui colormap picker would be very possible

### CODE REVIEW

File: src/cmap/utils.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+import numpy as np
+from typing_extensions import TypedDict
+
+
+def sineramp(
+    shape: tuple[int, int] | int = (256, 512),
+    amp: float = 0.05,
+    wavelen: float = 8,
+    power: float = 2,
+) -> np.ndarray:
```
Comment (alisterburt) [inline src/cmap/utils.py:24]: this is fun! did you port this yourself? I hadn't heard of Kovesi before, thanks for that!

**Review by alisterburt — APPROVED** [review]
This is gorgeous, and you put it together so quickly! Amazing work man - left a couple of small questions but overall this is just an awesome package, feels great to use and does exactly what you want. I learned a bunch of useful little tricks reading through the code too so thank you 🙂

**Comment by tlambert03** [comment]
so... actually I decided to make a new package after all: https://github.com/tlambert03/cmap

lots more stuff over there, including full coverage, and a lot of tests to make sure that this has parity with matplotlib's catalog and output of calling the colormap functions

File: src/cmap/colormap.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    from matplotlib.colors import LinearSegmentedColormap as MplLinearSegmentedColormap
+    from napari.utils.colormaps import Colormap as NapariColormap
+    from vispy.color import Colormap as VispyColormap
+
+    from .color import ValidColor
+
+
+class PColorMapper(Protocol):
+    @overload
+    def __call__(self, X: float) -> Color:
```
[the remark this answers is not in the document]
Comment (tlambert03) [reply src/cmap/colormap.py:34]: was what matlab did... removed over in cmap though

**Comment by alisterburt** [comment]
Love it, good call!

**Pull request closed without merging**